tests/plan/models/test_AnswerSet.py

```diff
@@ -0,0 +1,143 @@
+from django import test
+
+from easydmp.dmpt.models import ShortFreetextQuestion, BooleanQuestion, PositiveIntegerQuestion
+from easydmp.plan.models import Plan, AnswerSet, Answer
+from tests.dmpt.factories import TemplateFactory, SectionFactory
+
+
+class TestAnswerSetValidation(test.TestCase):
+
+    def setUp(self):
+        self.template = TemplateFactory()
+        self.plan = Plan(template=self.template, added_by_id=1, modified_by_id=1, valid=False)
+        self.plan.save()
+        self.section = SectionFactory.build(template=self.template)
+        self.section.save()
+        self.q1 = ShortFreetextQuestion(section=self.section, position=1)
+        self.q2 = BooleanQuestion(section=self.section, position=2)
+        self.q3 = PositiveIntegerQuestion(section=self.section, position=3)
+        self.q1.save()
+        self.q2.save()
+        self.q3.save()
+
+    def test_validate_answerset_data_ok(self):
+        as1 = AnswerSet(plan=self.plan, section=self.section, valid=False, data={
+            str(self.q1.id): {
+                "choice": "foo",
+                "notes": "n1"
+            },
+            str(self.q2.id): {
+                "choice": "Yes",
+                "notes": "n2"
+            },
+            str(self.q3.id): {
+                "choice": 1,
+                "notes": "n3"
+            }
+        })
+        as1.save()
```

Comment:
Hm what is codecov complaining about. Bah.

Reply:
@hmpf Just that one branch with the raise.